### PR TITLE
Auto-refresh cluster list on home screen

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/5446

This automatically refreshes the clusters list on the home screen. New data is loaded once every 30 seconds.

This includes some refactoring to make page visibility tracking re-usable.

### Preview

home-auto-refresh

Preview needs some patience, as the animation is a minutes long. Use [this link](http://recordit.co/myotVpsRZh) to scrub and fast-forward.

### CODE REVIEW

**Comment by marians** [comment]
For reviewers: The version is deployed on several test installations currently.

File: src/components/cluster/detail/cluster_detail_table.js [hunk 1]
```diff
@@ -12,10 +12,6 @@ import RefreshableLabel from '../../shared/refreshable_label';
 import ReleaseDetailsModal from '../../modals/release_details_modal';
 
 class ClusterDetailTable extends React.Component {
-  state = {
-    modifiedPropsPaths: [],
-  };
-
```
Comment (marians) [inline src/components/cluster/detail/cluster_detail_table.js]: Remove a left-over, unused bit of state

File: src/components/cluster/detail/view.js [hunk 2]
```diff
@@ -61,47 +62,18 @@ class ClusterDetailView extends React.Component {
...
-      // Handle page visibility change
-      document.addEventListener(
-        this.visibilityChange,
-        this.handleVisibilityChange,
-        false
-      );
-    }
-  };
+    this.visibilityTracker.addEventListener(this.handleVisibilityChange);
+  }
```
Comment (marians) [inline src/components/cluster/detail/view.js]: This moved into PageVisibilityTracker

File: src/components/home/index.js [hunk 3]
```diff
@@ -28,25 +40,48 @@ class Home extends React.Component {
...
-  clustersSortedById(clusters) {
-    return _.sortBy(clusters, 'id');
+  componentWillUnmount() {
+    this.visibilityTracker.removeEventListener(this.handleVisibilityChange);
   }
 
-  clusterIds(clusters) {
-    return this.clustersSortedById(clusters).map(cluster => cluster.id);
-  }
+  /**
```
Comment (marians) [inline src/components/home/index.js]: `clusterIds` and `clustersSortedById` were unused

File: src/components/cluster/detail/view.js (hunk 2 above)
Comment (oponder) [reply src/components/cluster/detail/view.js]: Nice refactor

**Pull request merged**